src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -810,22 +815,24 @@ private void BuildCookieCollectionFromDomainMatches(Uri uri, bool isSecure, int
                 PathList pathList;
                 lock (m_domainTable)
                 {
-                    if (!m_domainTable.TryGetValue(domainAttribute[i], out pathList))
+                    object pathListValue = m_domainTable[domainAttribute[i]];
+                    if (pathListValue == null)
                     {
                         continue;
                     }
+                    pathList = (PathList)pathListValue;
                 }
 
                 lock (pathList.SyncRoot)
                 {
-                    foreach (KeyValuePair<string, CookieCollection> pair in pathList)
+                    foreach (DictionaryEntry entry in pathList)
```

Comment:
This is going to allocate a boxed object for each entry in the dictionary. I wonder if `PathList.GetEnumerator` should be changed to return `IDictionaryEnumerator` instead of `IEnumerator`, so the allocations can be avoided while iterating through pathList.